macOS/UITests/FireWindowByDefaultTests.swift

```diff
@@ -0,0 +1,240 @@
+        continueAfterFailure = false
+
+        // Assume feature flag is on by default
+        app = XCUIApplication.setUp(featureFlags: ["openFireWindowByDefault": true])
+
+        dataClearingButton = app.buttons["PreferencesSidebar.dataClearingButton"]
+        generalButton = app.buttons["PreferencesSidebar.generalButton"]
+        openFireWindowByDefaultToggle = app.checkBoxes["PreferencesDataClearingView.openFireWindowByDefault"]
+        reopenAllWindowsFromLastSessionRadio = app.radioButtons["PreferencesGeneralView.stateRestorePicker.reopenAllWindowsFromLastSession"]
+
+        app.enforceSingleWindow()
+    }
+
+    override func tearDownWithError() throws {
+        // Reset both settings to default after each test
+        resetToDefaultSettings()
+        try super.tearDownWithError()
+    }
+
+    // MARK: - Test Cases
+
+    func testFireWindowByDefaultEnabled() {
+        // Navigate to Settings -> Data Clearing
+        navigateToDataClearingSettings()
+
+        // Turn on the 'Open Fire Window by Default' setting
+        if openFireWindowByDefaultToggle.value as? Bool == false {
+            openFireWindowByDefaultToggle.click()
+        }
+
+        // Verify the setting is enabled
+        XCTAssertTrue(openFireWindowByDefaultToggle.value as? Bool == true,
+                     "Open Fire Window by Default should be enabled")
+
+        // Close preferences
+        app.closePreferencesWindow()
+
+        // Test CMD+N opens Fire Window
+        app.typeKey("n", modifierFlags: .command)
+        assertFireWindowOpened()
+
+        // Close the Fire Window
+        app.closeWindow()
+
+        // Test CMD+SHIFT+N opens Normal Window
+        app.typeKey("n", modifierFlags: [.command, .shift])
+        assertNormalWindowOpened()
+    }
+
+    func testFireWindowByDefaultDisabled() {
+        // Navigate to Settings -> Data Clearing
+        navigateToDataClearingSettings()
+
+        // Turn off the 'Open Fire Window by Default' setting
+        if openFireWindowByDefaultToggle.value as? Bool == true {
+            openFireWindowByDefaultToggle.click()
+        }
+
+        // Verify the setting is disabled
+        XCTAssertTrue(openFireWindowByDefaultToggle.value as? Bool == false,
+                     "Open Fire Window by Default should be disabled")
+
+        // Close preferences
+        app.closePreferencesWindow()
+
+        // Test CMD+N opens Normal Window
+        app.typeKey("n", modifierFlags: .command)
+        assertNormalWindowOpened()
+
+        // Close the Normal Window
+        app.closeWindow()
+
+        // Test CMD+SHIFT+N opens Fire Window
+        app.typeKey("n", modifierFlags: [.command, .shift])
+        assertFireWindowOpened()
+    }
+
+    func testFireWindowByDefaultSessionRestoreInteraction() {
+        // First, enable session restore in General preferences
+        navigateToGeneralSettings()
+
+        if reopenAllWindowsFromLastSessionRadio.isSelected == false {
```

Comment:
please use `app.preferencesSetRestorePreviousSession` defined in `XCUIApplicationExtension`